CMakeLists.txt

```diff
@@ -1,39 +1,76 @@
 cmake_minimum_required(VERSION 2.8)
 
-set(PROJ_CJSON cJSON)
+set(CJSON_VERSION_MAJOR 1)
+set(CJSON_VERSION_MINOR 0)
+set(CJSON_VERSION_PATCH 0)
+set(CJSON_VERSION_SO 1)
+set(CJSON_UTILS_VERSION_SO 1)
+set(CJSON_VERSION "${CJSON_VERSION_MAJOR}.${CJSON_VERSION_MINOR}.${CJSON_VERSION_PATCH}")
+
+add_compile_options(-ansi -pedantic -Wall -Wextra -Werror -Wstrict-prototypes -Wwrite-strings)
```

Comment:
I will make this conditional so it is only applied when GCC or Clang is detected.